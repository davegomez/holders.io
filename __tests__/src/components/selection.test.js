import React from 'react'
import Selection from '../../../src/components/selection'
import renderer from 'react-test-renderer'

test('Selection component', () => {
  const component = renderer.create(
    <Selection
      label='foo'
      value='bar'
      options={[ { value: 'foo', label: 'bar'} ]}
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
