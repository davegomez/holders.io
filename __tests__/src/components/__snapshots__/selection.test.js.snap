// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection component 1`] = `
<label
  htmlFor="foo"
>
  foo
  <div
    className="Select Select--single is-clearable is-searchable"
    style={undefined}
  >
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-foo--value"
      >
        <div
          className="Select-placeholder"
        >
          Select...
        </div>
        <div
          className="Select-input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-activedescendant="react-select-foo--value"
            aria-describedby={undefined}
            aria-expanded="false"
            aria-haspopup="false"
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            role="combobox"
            style={
              Object {
                "boxSizing": "content-box",
                "width": "5px",
              }
            }
            tabIndex={undefined}
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </span>
      <span
        aria-label="Clear value"
        className="Select-clear-zone"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        title="Clear value"
      >
        <span
          className="Select-clear"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&times;",
            }
          }
        />
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
</label>
`;
